cloud/services/inboundnatrules/inboundnatrules.go

```diff
@@ -41,14 +41,19 @@ func (s *Service) Reconcile(ctx context.Context) error {
 		if lb.LoadBalancerPropertiesFormat == nil || lb.FrontendIPConfigurations == nil || lb.InboundNatRules == nil {
 			return errors.Errorf("Could not get existing inbound NAT rules from load balancer %s properties", to.String(lb.Name))
 		}
+		exists := false
```

Comment:
style: when I find myself using this pattern, I often prefer refactoring the loop logic into a function so that I can use return + continue from the inner loop:
```go
// would just be if fn() {} but verbose for illustration
if exists() {
    // now this breaks the outer loop
    continue
}

func exists() bool {
    for _, foo := range bar {
        if condition(foo) {
            return true
        }
    }
    return false
}
```

Reply:
Usually this is better but in this case it's doing a bit more that that, we're also putting the ports in a map to be able to get an unused port later so it would mean either the exists() func has to be overloaded and return a ports map as well or we have to loop through the rules twice. I think I prefer keeping it as-is.

Reply:
https://diffy.org/diff/3ezc7mo6trys9n74ve2rv5z5mi would be one way to do it

Not a huge deal either way

Reply:
how about https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/893/commits/86b9cf769e5261883b7517995d305c66bcb71627

I will squash if you like it

Reply:
yep, nice